### ISSUE TITLE
RuboCop: Lint/Syntax: unexpected token tEQ error when line starts with the `==`

### ISSUE THREAD

**Issue by K-S-A** [sds/haml-lint#419]
Example haml file:

```haml
-# `/test.html.haml` 
%pre== #{'Hello'} World
%pre
  == #{'Hello'} World
```

It is completely valid (https://github.com/haml/haml/blob/v6.1.1/lib/haml/parser.rb#L288) and should result in the following html output:

```html
<pre>Hello World</pre>
<pre>Hello World</pre>
```

Though, the `haml-lint` check fails with an error:

```console
/test.html.haml:3 [E] RuboCop: Lint/Syntax: unexpected token tEQL
(Using Ruby 3.1 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

1 file inspected, 1 lint detected
```

The reason is that the `== #{'Hello'} World` line is passed to the RuboCop in a form of `= #{'Hello'} World`.

The issue is traced down to this line https://github.com/sds/haml-lint/blob/v0.47.0/lib/haml_lint/ruby_extraction/chunk_extractor.rb#L92. The `/(=[ \t]?)/` regexp replaced only the `= ` part and left the first `=` in place.

The proposed fix is:

```ruby
lines[0] = lines[0].sub(/(={1,2}[ \t]?)/, '')
```

I'll be happy to submit a PR for this.

**Comment by MaxLap** [comment]
I've never seen `==`. Took me a while to find anything about it... Found it in the changelog:

```
In Haml 2.0, you could use `==` to interpolate Ruby code
within a line of text using `#{}`.
In Haml 2.2, the `==` is unnecessary;
`#{}` can be used in any text.
```

So basically it's a feature that has no use since HAML 2.2. You could simply remove the `==`.

Your proposed fix will not work because the handling is totally different. After a `=`, valid Ruby code is expected.
After a `==`, it's text with possible indentation.

I'll take care of the PR since it's a bit more involved.

**Comment by MaxLap** [comment]
Here's the 500 lines PR (it's mostly tests xD) if your are interested: #420
